src/node/rpc/node_frontend.h

```diff
@@ -409,6 +416,93 @@ namespace ccf
         .set_openapi_hidden(true)
         .install();
 
+      auto promote = [this](auto& args, const nlohmann::json& params) {
+        const auto in = params.get<PromoteNodeToTrusted::In>();
+
+        if (
+          !this->context.get_node_state().is_part_of_network() &&
+          !this->context.get_node_state().is_part_of_public_network() &&
+          !this->context.get_node_state().is_reading_private_ledger())
+        {
+          return make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            "Target node should be part of network to promote nodes");
+        }
+
+        if (!consensus)
+        {
+          return make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            "No configured consensus");
+        }
+
+        if (
```

Comment:
This should already be handled by the HTTP endpoint framework as the endpoint is a write endpoint.